app/gui/src/dashboard/providers/SessionProvider/SessionProvider.tsx

```diff
@@ -289,6 +289,21 @@ export function SessionProvider(props: SessionProviderProps) {
     }
   }, [session.data, saveAccessTokenEventCallback])
 
+  React.useEffect(() => {
+    queryClient.getQueryCache().config.onError = (error, query) => {
+      if (error instanceof NotAuthorizedError) {
```

Comment:
This might execute multiple refreshes at the time, I assume.

Reply:
yup. i didn't want to introduce additional complexity, although i can kinda see a simply way to solve it?

Reply:
This can't be simple, per the article above, unfortunately

Reply:
is there a reason this doesn't work (at least theoretically)? https://github.com/enso-org/enso/pull/13122/commits/cb83c8a35a69719417c6e6a9361eb08bd1ec3174

Reply:
asking because this PR does the refetches completely differently to the other article (i'm really unsure about manually creating a queue)